Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -251,5 +263,516 @@ namespace AZ
+            }
+
+            // Tooltip and block highlighting
+            if (ImGui::IsMouseHoveringRect(startPoint, endPoint) && ImGui::IsWindowHovered())
+            {
+                // Open function statistics map on click
+                if (ImGui::IsMouseClicked(ImGuiMouseButton_Left))
+                {
+                    const GroupRegionName* key = block.m_groupRegionName;
+                    m_regionStatisticsMap[key].m_draw = true;
+                }
+
+                // Hovering outline
+                drawList->AddRect(startPoint, endPoint, ImGui::GetColorU32({ 1, 1, 1, 1 }), 0.0, 0, 1.5);
+
+                ImGui::BeginTooltip();
+                ImGui::Text("%s::%s", block.m_groupRegionName->m_groupName, block.m_groupRegionName->m_regionName);
+                ImGui::Text("Execution time: %.3f ms", CpuProfilerImGuiHelper::TicksToMs(block.m_endTick - block.m_startTick));
+                ImGui::Text("%lld => %lld", block.m_startTick, block.m_endTick);
+                ImGui::EndTooltip();
+            }
+        }
+
+        inline ImU32 ImGuiCpuProfiler::GetBlockColor(const TimeRegion& block)
+        {
+            // Use the GroupRegionName pointer a key into the cache, equal regions will have equal pointers
+            const GroupRegionName* key = block.m_groupRegionName;
+            if (m_regionColorMap.contains(key)) // Cache hit
+            {
+                return ImGui::GetColorU32(m_regionColorMap[key]);
+            }
+
+            // Cache miss, generate a new random color
+            std::mt19937 mt(m_rd());
+            std::uniform_real_distribution<float> dis(.1, .9);
+            const ImVec4 randomColor = { dis(mt), dis(mt), dis(mt), .8 };
+            m_regionColorMap.emplace(key, randomColor);
+            return ImGui::GetColorU32(randomColor);
+        }
+
+        inline void ImGuiCpuProfiler::DrawThreadSeparator(u64 baseRow, u64 maxDepth)
+        {
+            const ImU32 red = ImGui::GetColorU32({ 1, 0, 0, 1 });
+
+            auto [wx, wy] = ImGui::GetWindowPos();
+            wy -= ImGui::GetScrollY();
+            const float windowWidth = ImGui::GetWindowWidth();
+            const float boundaryY = wy + (baseRow + maxDepth + 1) * RowHeight - 5;
+
+            ImGui::GetWindowDrawList()->AddLine({ wx, boundaryY }, { wx + windowWidth, boundaryY }, red, 2.0f);
+        }
+
+        inline void ImGuiCpuProfiler::DrawThreadLabel(u64 baseRow, AZStd::thread_id threadId)
+        {
+            auto [wx, wy] = ImGui::GetWindowPos();
+            wy -= ImGui::GetScrollY();
+            const AZStd::string threadIdText = AZStd::string::format("Thread %zu", static_cast<size_t>(threadId.m_id));
+
+            ImGui::GetWindowDrawList()->AddText({ wx + 10, wy + baseRow * RowHeight + 5 }, IM_COL32_WHITE, threadIdText.c_str());
+        }
+
+        inline void ImGuiCpuProfiler::DrawRegionStatistics()
+        {
+            for (auto& [groupRegionName, stat] : m_regionStatisticsMap)
```

Comment:
I think this can be 'const' auto&

Reply:
`Stat` needs to be mutable so that we can pass it to ImGui::Begin on line 622